src/compiler/declarationEmitter.ts

```diff
@@ -396,6 +398,12 @@ namespace ts {
                 }
             }
 
+            function emitTypePredicate(type: TypePredicateNode) {
+                writeEntityName(type.parameterName);

```

Comment:
>What is the difference between writeEntityName and emitEntityName?

`emitEntityName` checks for error. `writeEntityName` doesn't check.

>Can you encode this difference into the names of the respective functions?

Then I need to change `emitEntityName` which break naming consistency with other emit functions.